cmake/modules/compiler_config.cmake

```diff
@@ -47,33 +47,22 @@ function(SET_CACHED_VARIABLE var)
   set(_cached_vars ${_cached_vars} CACHE INTERNAL "")
 endfunction(SET_CACHED_VARIABLE)
 
-############################################################################
-## GCC specific options ####################################################
-############################################################################
+# Default flags to compiler when build user-space programs.
+# Should come before enabling language.
 
-if(CMAKE_COMPILER_IS_GNUCC)
-  # Default flags to compiler when build user-space programs.
-  # Should come before enabling language.
-  set(CMAKE_C_FLAGS_DEBUG "-g -O0 -Wall -Wextra"
-    CACHE STRING "Compiler flags for debug builds.")
-  set(CMAKE_C_FLAGS_RELWITHDEBINFO "-g -Wall -Wextra"
-    CACHE STRING "Compiler flags for Release With Debug Info builds.")
-  set(CMAKE_C_FLAGS_RELEASE "-Wall"
-    CACHE STRING "Compiler flags for release builds.")
-  set(CMAKE_CXX_FLAGS_DEBUG "-g -O0 -Wall -Wextra"
-    CACHE STRING "C++ compiler flags for debug builds.")
-  set(CMAKE_CXX_FLAGS_RELWITHDEBINFO "-g -Wall -Wextra"
-    CACHE STRING "C++ compiler flags for Release with Debug Info builds.")
-  set(CMAKE_CXX_FLAGS_RELEASE "-Wall"
-    CACHE STRING "C++ compiler flags for release builds.")
-endif(CMAKE_COMPILER_IS_GNUCC)
+#TODO enable -Werror
+
+set(CMAKE_C_FLAGS_DEBUG            "-g -O0 -Wall -Wextra")
+set(CMAKE_CXX_FLAGS_DEBUG          "-g -O0 -Wall -Wextra")
 
-# Recommend enabling Werror. Can't enable it yet because we are not warning clean
-#set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Wall -Wextra -Werror")
-#set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Wall -Wextra -Werror")
+set(CMAKE_C_FLAGS_RELEASE          "-Wall -Wextra")
+set(CMAKE_CXX_FLAGS_RELEASE        "-Wall -Wextra")
 
-set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -Wall -Wextra")
-set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Wall -Wextra")
+set(CMAKE_C_FLAGS_RELWITHDEBINFO   "-g -Wall -Wextra")
```

Comment:
Needs `-O2` (or whatever optimization level we want for release)

Reply:
done